src/einsteinpy/symbolic/ricci.py

```diff
@@ -96,3 +154,74 @@ def symbols(self):
 
         """
         return self.syms
+
+
+class RicciScalar:
+    """
+    Class for defining Ricci Scalar
+    """
+
+    def __init__(self, metric_tensor, ricci_tensor):
+        """
+        Constructor and Initializer
+
+        Parameters
+        ----------
+        metric_tensor : ~einsteinpy.symbolic.metric.MetricTensor
+            Metric Tensor from which Ricci Scalar to be calculated
+        ricci_tensor : ~einsteinpy.symbolic.metric.RicciTensor
+            Ricci Tensor from which Ricci Scalar to be calculated
+
+        Raises
+        ------
+        TypeError
+            Raised when metric_tensor is not a MetricTensor or/and ricci_tensor is not a RicciTensor
+
+        """
+        if isinstance(metric_tensor, MetricTensor) and isinstance(
+            ricci_tensor, RicciTensor
+        ):
+            self.ricci_scalar = self.calculate_ricci_scalar(metric_tensor, ricci_tensor)
+
+        else:
+            raise TypeError(
+                "metric_tensor should be a MetricTensor class and ricci_tensor should be a RicciTensor"
+            )
```

Comment:
Remove this codebase
And add the following:
```
self._order = 0
self._parent_metric = parent_metric
self._expr = expression
if isinstance(syms, (list, tuple)):
    self.syms = syms
    self.dims = len(self.syms)
else:
    raise TypeError("syms should be a list or tuple")
```
We would have classmethods like `from_metric`, `from_christoffel`, `from_riemann`, `from_ricci` for deriving Ricci Scalar from other quantities

Reply:
Okey. That totally changes the focus I was giving him. I start doing these classmethod for the next push